### DISCUSSION TITLE
OSV: further refinement needed to vulnerability severity feature

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#22239] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

We've been using https://github.com/renovatebot/renovate/pull/21939 for several days now across our org, and I've observed the following two scenarios that need additionally handling (happy to do so via a PR once we reach consensus on way forward)

I updated https://github.com/setchy/renovate-demo-osv-severity to demonstrate the following

Observation 1 - `MEDIUM` + `MODERATE`

- Both `MEDIUM` and `MODERATE` can be valid severity ratings
  - `MEDIUM`: https://github.com/setchy/renovate-demo-osv-severity/pull/7
  - `MODERATE`: https://github.com/setchy/renovate-demo-osv-severity/pull/8
  
- Should we 
  - support both of these severity types (ie: the native severity rating from the osv database)?
  - Or, should we normalize these to a common value?

- My recommendation would be to support both, but keen to hear input from others.

Observation 2 - `Unknown severity`

- For updates that have an Unknown severity, we currently don't set a `vulnerabilitySeverity`
- Example: https://github.com/setchy/renovate-demo-osv-severity/pull/9
- Proposal: set `vulnerabilitySeverity` to `UNKNOWN` and update sort logic accordingly

**Comment by setchy** [comment]
@JamieMagee @viceice @secustor - appreciate your thoughts on how best to handle the above 😄

**Reply by JamieMagee** [reply]
For your first point, I agree that supporting both might be easier. Extending [this map][1] to include `MEDIUM` is simple.

For your second point, I agree that adding `UNKNOWN` is the best approach, but where do you sort them? My gut feeling is higher than `CRITICAL`. If we were to sort them lower than `LOW`, or in the middle at `MEDIUM`/`MODERATE` there might be a false sense of security.

[1]: https://github.com/renovatebot/renovate/blob/f86fa3af8f0cfeb6c4c22390bbbeb39f511ccb0b/lib/util/vulnerability/utils.ts#L1-L6

**Reply by setchy** [reply]
Excellent - here's one i prepared earlier https://github.com/renovatebot/renovate/pull/22257
